src/main/java/de/tum/in/www1/exerciseapp/web/rest/ExerciseResource.java

```diff
@@ -121,7 +117,7 @@ public ExerciseResource(ExerciseRepository exerciseRepository, ExerciseService e
      * @throws URISyntaxException if the Location URI syntax is incorrect
      */
     @PutMapping("/exercises")
-    @PreAuthorize("hasAnyRole('ADMIN', 'TA')")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
     @Timed
     //TODO: test if it still works with abstract entity in body
     public ResponseEntity<Exercise> updateExercise(@RequestBody Exercise exercise) throws URISyntaxException {
```

Comment:
This should only be possible for TAs and instrcutors of this particular course (and admins)

Reply:
Same here. Edit exercise feature is using PUT /api/programming-exercises. I wasn't able to identify the use of this method.

Reply:
Done.